### ISSUE TITLE
Crashes on Windows using Python 3.12.0b3 when `cffi` is installed

### ISSUE THREAD

**Issue by bastimeyer** [Legrandin/pycryptodome#754]
Hi, I just noticed that `pycryptodome` caused a Python crash after adding a Windows cp312 CI runner to our project. Our Linux CI runner for cp312 has been working fine for the past weeks.

Since there were no error logs, I dug a bit deeper and checked in a local VM. Apparently, the crash is caused when `cffi` is installed in the same Python environment. `cffi` is a transitive dependency of our project, required by `trio`.

Installing `pycryptodome` from the master branch results in the same issue. cp311 works fine.

```
$ python -VV
Python 3.12.0b3 (tags/v3.12.0b3:f992a60, Jun 20 2023, 12:25:40) [MSC v.1936 64 bit (AMD64)]

$ pip list
Package Version
------- -------
pip     23.1.2

$ pip install pycryptodome
Collecting pycryptodome
  Using cached pycryptodome-3.18.0-cp35-abi3-win_amd64.whl (1.7 MB)
Installing collected packages: pycryptodome
Successfully installed pycryptodome-3.18.0

$ python -c 'from Crypto.Cipher import AES'
$ echo $?
0

$ pip install cffi
Collecting cffi
  Using cached cffi-1.15.1-cp312-cp312-win_amd64.whl
Collecting pycparser (from cffi)
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Installing collected packages: pycparser, cffi
Successfully installed cffi-1.15.1 pycparser-2.21

$ python -c 'from Crypto.Cipher import AES'
$ echo $?
127
```

I have no experience with Python C-extensions, so I'm stuck here. I don't know if this is an issue with `cffi` or with `pycryptodome`, hence why I'm opening this issue here first.

Thanks!

**Comment by Legrandin** [comment]
No tests have been done with Python 3.12 so far since it's not released yet.
But could please also check with cp311?

**Comment by bastimeyer** [comment]
> cp311 works fine.

**Comment by Legrandin** [comment]
There is indeed a problem (in Windows), but I also see that the [`cffi` project](https://foss.heptapod.net/pypy/cffi/-/issues/562) had to make some recent low-level changes to be compatible with Python 3.12. So, I would like to look at this issue only after the next `cffi` release is available.

**Comment by Legrandin** [comment]
I have traced the crash down to a bug in cffi, but I am now waiting to get access (!) to their tracker to notify the maintainers.

**Comment by Legrandin** [comment]
Still no access, so I notified the maintainers via Google Groups:

https://groups.google.com/u/1/g/python-cffi/c/oZkOIZ_zi5k

**Comment by Legrandin** [comment]
Fixed with release v3.19 (we simply don't use cffi on Windows with Python 3.12, until a new release is out).
